build/images/Dockerfile.build.agent.coverage

```diff
@@ -41,4 +41,7 @@ COPY --from=antrea-build /antrea/bin/antrea-agent-coverage /usr/local/bin/
 COPY --from=antrea-build /antrea/bin/antrea-cni /usr/local/bin/
 COPY --from=antrea-build /antrea/bin/antctl /usr/local/bin/
 COPY --from=antrea-build /antrea/bin/antctl-coverage /usr/local/bin/
-COPY --from=antrea-build /antrea/test/e2e/coverage/agent-arg-file /
+COPY --from=antrea-build /antrea/test/e2e/utils/run_cov_binary.sh /
+RUN mkdir -p /tmp/coverage
```

Comment:
Looks like 1 is not an option. I also realized that this may be a bit of an issue when we run `antctl` with coverage enabled as a Pod in `test/e2e/antctl_test.go`.

So thinking about it more, I think the best option would be to have the following in the Dockerfiles:
```
# This environment variable will persist when running the container, but can also be overwritten if needed
ENV GOCOVERDIR=/tmp/coverage
RUN mkdir -p $GOCOVERDIR
```

This way:
- we keep things consistent: we create the directory and set the env variable in the same place, no need to duplicate the value
- we no longer need to set `GOCOVERDIR` in the yaml manifests, which will reduce verbosity